paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/PaymentOption.kt

```diff
@@ -1,20 +1,52 @@
 package com.stripe.android.paymentsheet.model
 
+import android.graphics.drawable.Drawable
 import androidx.annotation.DrawableRes
 
 /**
  * The customer's selected payment option.
  */
-data class PaymentOption(
+data class PaymentOption
+@Deprecated("Not intended for public use.")
+constructor(
     /**
      * The drawable resource id of the icon that represents the payment option.
      */
-    @DrawableRes val drawableResourceId: Int,
+    @Deprecated("Please use iconDrawable instead.")
+    @DrawableRes
+    val drawableResourceId: Int,
 
     /**
      * A label that describes the payment option.
      *
      * For example, "····4242" for a Visa ending in 4242.
      */
     val label: String
-)
+) {
+    // These aren't part of the primary constructor in order to maintain binary compatibility.
+    internal var lightThemeIconUrl: String? = null
+        private set
+
+    internal var darkThemeIconUrl: String? = null
+        private set
+
+    private var imageLoader: suspend (PaymentOption) -> Drawable = {
+        throw IllegalStateException("Must pass in an image loader to use iconDrawable.")
+    }
+
+    internal constructor(
+        @DrawableRes drawableResourceId: Int,
+        label: String,
+        lightThemeIconUrl: String?,
+        darkThemeIconUrl: String?,
+        imageLoader: suspend (PaymentOption) -> Drawable
+    ) : this(drawableResourceId, label) {
+        this.lightThemeIconUrl = lightThemeIconUrl
+        this.darkThemeIconUrl = darkThemeIconUrl
+        this.imageLoader = imageLoader
+    }
+
+    suspend fun iconDrawable(): Drawable {
```

Comment:
Nit: Can we name this `loadIcon()` or `fetchIcon()` or something similar?

Reply:
Yeah, I'll start with `fetchIcon`. After talking with wooj, we need to do an API review for this change, and this will be part of that, so it might change again.